addons/im_livechat/models/discuss_channel.py

```diff
@@ -310,14 +313,14 @@ def _get_visitor_leave_message(self, operator=False, cancel=False):
     def _close_livechat_session(self, **kwargs):
         """ Set deactivate the livechat channel and notify (the operator) the reason of closing the session."""
         self.ensure_one()
-        if self.livechat_active:
+        if not self.livechat_end_dt:
             member = self.channel_member_ids.filtered(lambda m: m.is_self)
             if member:
                 # sudo: discuss.channel.rtc.session - member of current user can leave call
                 member.sudo()._rtc_leave_call()
             # sudo: discuss.channel - visitor left the conversation, state must be updated
-            self.sudo().livechat_active = False
-            self.sudo()._bus_send_store(Store(self, "livechat_active"))
+            self.sudo().livechat_end_dt = fields.Datetime.now()
+            self.sudo()._bus_send_store(self, "livechat_end_dt")
```

Comment:
`sudo` before `_bus_send_store` looks unnecessary (`self` is never used in there except for `_bus_channel` but that method is not overridden in channel anyway). If anything, the self param of the function should be having sudo.